client-apis/encrypt.md

```diff
@@ -113,68 +161,168 @@ This document refers to the output of the key derivation algorithm as the derive
+in [sequence number](#.../data-format/message-body/md#sequence-number)
+order is equal to the input plaintext which has been encrypted so far.
+
+If the caller has not indicated an end to the input plaintext yet
+and there are not enough input plaintext bytes available to create a new
+[regular frame](#../data-format/message-body.md#regular-frame),
+then this operation MUST wait until either enough plaintext bytes become available
+or the caller indicates an end to the plaintext.
+
+If the caller has not indicated an end to the input plaintext yet
+but there is enough plaintext available to construct a new regular frame,
+then this operation MUST [construct a regular frame](#construct-a-frame)
+with the avilable plaintext.
+
+If the caller has indicated an end to the input plaintext
+and there is exactly enough plaintext bytes available to create one regular frame,
+then this operation MUST [construct either a final frame or regular frame](#construct-a-frame)
+with the remaining plaintext.
+If they construct a regular frame, they MUST also construct a empty final frame.
+
+If the caller has indicated an end to the input plaintext
+and there are not enough input plaintext bytes available to create a new regular frame,
+then this operation MUST [construct a final frame](#construct-a-frame)
+with the remaining plaintext.
+
+### Construct a frame
+
+To construct a regular or final frame that represents the next frame in the encrypted message's body,
+this operation MUST calculate encrypted content and an authentication tag using the
+[authenticated encryption algorithm](../framework/algorithm-suites.md#encryption-algorithm)
+specified by the [algorithm suite](../framework/algorithm-suites.md),
+with the following inputs:
+
+- The AAD is the serialized [message body AAD](../data-format/message-body-aad.md),
+  constructed as follows:
+  - The [message ID](../data-format/message-body-aad.md#message-id) is the same as the
+    [message ID](../data-frame/message-header.md#message-id) serialized in the header of this message.
+  - The [Body AAD Content](../data-format/message-body-aad.md#body-aad-content) depends on
+    whether the thing being encrypted is a regular frame or final frame.
+    Refer to [Message Body AAD](../data-format/message-body-aad.md) specification for more information.
+  - The [sequence number](../data-format/message-body-aad.md#sequence-number) is the sequence
+    number of the frame being encrypted.
+    If this is the first frame sequentially, this value MUST be 1.
+    Otherwise, this value MUST be 1 greater than the value of the sequence number
+    of the previous frame.
+  - The [content length](../data-format/message-body-aad.md#content-length) MUST have a value
+    equal to the length of the plaintext being encrypted.
+- The IV is the [sequence number](../data-format/message-body-aad.md#sequence-number)
+  used in the message body AAD above.
```

Comment:
As noted above padded with 0 to 12 bytes. (Which is also specified by the algorithm suite...)

Reply:
Updated.